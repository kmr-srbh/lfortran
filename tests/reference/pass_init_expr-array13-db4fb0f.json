{
    "basename": "pass_init_expr-array13-db4fb0f",
    "cmd": "lfortran --pass=init_expr --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/array13.f90",
    "infile_hash": "9a8fc967d5ae775229d7d6e3fff0a1927b7bc3ec0b5209bedf864ff8",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "pass_init_expr-array13-db4fb0f.stdout",
    "stdout_hash": "140ab49fb9c95aa218df4d515aa60d7c5dd54c21d6cc621e361fb8be",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}