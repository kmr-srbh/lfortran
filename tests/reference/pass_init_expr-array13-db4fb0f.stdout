(TranslationUnit
    (SymbolTable
        1
        {
            array13:
                (Program
                    (SymbolTable
                        2
                        {
                            __1_k:
                                (Variable
                                    2
                                    __1_k
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                ),
                            numbers:
                                (Variable
                                    2
                                    numbers
                                    []
                                    Local
                                    ()
                                    ()
                                    Save
                                    (Array
                                        (Integer 4)
                                        [((IntegerConstant 1 (Integer 4) Decimal)
                                        (IntegerConstant 5 (Integer 4) Decimal))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                )
                        })
                    array13
                    []
                    [(Assignment
                        (Var 2 __1_k)
                        (ArrayBound
                            (Var 2 numbers)
                            (IntegerConstant 1 (Integer 4) Decimal)
                            (Integer 4)
                            LBound
                            ()
                        )
                        ()
                    )
                    (Assignment
                        (ArrayItem
                            (Var 2 numbers)
                            [(()
                            (Var 2 __1_k)
                            ())]
                            (Integer 4)
                            RowMajor
                            ()
                        )
                        (IntegerConstant 5 (Integer 4) Decimal)
                        ()
                    )
                    (Assignment
                        (Var 2 __1_k)
                        (IntegerBinOp
                            (Var 2 __1_k)
                            Add
                            (IntegerConstant 1 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (Assignment
                        (ArrayItem
                            (Var 2 numbers)
                            [(()
                            (Var 2 __1_k)
                            ())]
                            (Integer 4)
                            RowMajor
                            ()
                        )
                        (IntegerConstant 4 (Integer 4) Decimal)
                        ()
                    )
                    (Assignment
                        (Var 2 __1_k)
                        (IntegerBinOp
                            (Var 2 __1_k)
                            Add
                            (IntegerConstant 1 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (Assignment
                        (ArrayItem
                            (Var 2 numbers)
                            [(()
                            (Var 2 __1_k)
                            ())]
                            (Integer 4)
                            RowMajor
                            ()
                        )
                        (IntegerConstant 2 (Integer 4) Decimal)
                        ()
                    )
                    (Assignment
                        (Var 2 __1_k)
                        (IntegerBinOp
                            (Var 2 __1_k)
                            Add
                            (IntegerConstant 1 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (Assignment
                        (ArrayItem
                            (Var 2 numbers)
                            [(()
                            (Var 2 __1_k)
                            ())]
                            (Integer 4)
                            RowMajor
                            ()
                        )
                        (IntegerConstant 1 (Integer 4) Decimal)
                        ()
                    )
                    (Assignment
                        (Var 2 __1_k)
                        (IntegerBinOp
                            (Var 2 __1_k)
                            Add
                            (IntegerConstant 1 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (Assignment
                        (ArrayItem
                            (Var 2 numbers)
                            [(()
                            (Var 2 __1_k)
                            ())]
                            (Integer 4)
                            RowMajor
                            ()
                        )
                        (IntegerConstant 3 (Integer 4) Decimal)
                        ()
                    )
                    (Assignment
                        (Var 2 __1_k)
                        (IntegerBinOp
                            (Var 2 __1_k)
                            Add
                            (IntegerConstant 1 (Integer 4) Decimal)
                            (Integer 4)
                            ()
                        )
                        ()
                    )]
                )
        })
    []
)
